### DISCUSSION TITLE
Why vfile-message dependence is self with yarn @type?

### DISCUSSION THREAD

**Discussion by lihsai0** [vfile/vfile#61] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/17806506/103353675-64425600-4ae4-11eb-9861-7557be2aea23.png)

I am using [gatsby-plugin-ts](https://github.com/d4rekanguok/gatsby-typescript/tree/master/packages/gatsby-plugin-ts) and [gatsby-theme-blog](https://github.com/gatsbyjs/themes/tree/master/packages/gatsby-theme-blog).

Got an Error when tsc: lost /node_modules/@types/vfile/index.d.ts

I am not sure is it a npm publish mistake? or it's a mistake by yarn?

P.S. npm works.

**Comment by ChristianMurphy** [comment, marked as answer]
the `@types/vfile-message` package has been deprecated/replaced by `vfile-message` having it's own native typings.
Definitely typed as part of the deprecation process, redirects the types to the actual package https://github.com/DefinitelyTyped/DefinitelyTyped/blob/3ca6e2420bb5dbef73e971871890e5f47f1f0d79/notNeededPackages.json#L4844-L4848
any package still depending on `@types/vfile-message` should remove the dependency, and update to `vfile-message` latest if possible.

As for the specific error, it sound like a problem with yarn.

**Reply by lihsai0** [reply]
Thank you!

I think I found others problem. It's may a tsc config problem:

When tsc compile `node_modules/vfile/`, tsc should use `node_modules/vfile/node_modules/vfile-message/types/index.d.ts` not `node_modules/@types/vfile-message/index.d.ts`

~~But I have no idea how to make it.~~ Anyway sorry for bothering you and thanks again!

**Reply by lihsai0** [reply]
Oh! I'm stupit!

These change in `tsconfig.json` make it:

``` diff
- "typeRoots": ["./types", "./node_modules/@types"]
+ "typeRoots": ["types"]
```
